repository/src/main/java/org/apache/atlas/util/AtlasMetricsUtil.java

```diff
@@ -251,4 +257,48 @@ private String millisToTimeDiff(long msDiff) {
     private String millisToTimeStamp(long ms) {
         return simpleDateFormat.format(ms);
     }
+
+    public void setSuccessfulMetrics(HookNotification.HookNotificationType type, long timestamp) {
```

Comment:
Consider replacing methods setSuccessfulMetrics() and setFailedMetrics() with the following method:
public void onNotificationProcessingComplete(HookNotificationType type, boolean isSuccess) {
  long startTimeMs = RequestContext.get().getRequestTime();
  long timeTakenMs = System.currentTimeMillis() - startTimeMs;

  // ..
}

Reply:
This change includes moving business logic to util class. I will improve the algo itself later. And for unit testing purpose, I kept $long timestamp$ as a parameter so I could fake the testing data. Thanks.